### PR TITLE
Dustin decker/max concurrent jobs

### LINKED ISSUE #53: [FEATURE] - Maximum concurrent jobs
### Is your feature request related to a problem? Please describe

When running resource intensive jobs, it may be desirable to limit how many are executed at once.

### Describe the solution you'd like

I would like to be able to configure maximum running jobs.

### Describe alternatives you've considered

I can probably do a hacky implementation of this with the Locker feature.

### PR DESCRIPTION
**Pull request by JohnRoesler**
### What does this do?
- Adds the ability to limit the maximum concurrently running jobs on the scheduler
- Adds modes for the max concurrency, of reschedule or wait

### Which issue(s) does this PR fix/relate to?

resolves #53 
closes #63 

### Have you included tests for your changes?
👍 

### Did you document any new/modified functionality?

- [x] Updated `example_test.go`
- [ ] Updated `README.md`

### CODE REVIEW

**Comment by JohnRoesler** [comment]
if ya'll wouldn't mind taking a look and let me know what you think of this implementation, I would greatly appreciate it! (i tagged everyone involved in the other open PR)

@dustin-decker, @temamagic, @farwydi, @Streppel, @arjunmahishi

File: executor.go [hunk 1]
```diff
@@ -25,6 +46,26 @@ func (e *executor) start() {
...
+						switch e.limitMode {
+						case RescheduleMode:
+							return
+						case WaitMode:
+							for {
+								if !e.maxRunningJobs.TryAcquire(1) {
+									continue
+								} else {
+									break
+								}
```
Comment (Streppel) [inline executor.go:61]: wdyt?
```suggestion
								if e.maxRunningJobs.TryAcquire(1) {
									break
								}
```

File: executor.go [hunk 2]
```diff
@@ -25,6 +46,26 @@ func (e *executor) start() {
 			go func() {
 				defer wg.Done()
 
+				if e.maxRunningJobs != nil {
+					if !e.maxRunningJobs.TryAcquire(1) {
+
+						switch e.limitMode {
```
Comment (Streppel) [inline executor.go:52]: really liked this approach

**Review by Streppel — APPROVED** [review]

File: executor.go [hunk 3]
```diff
@@ -25,6 +46,26 @@ func (e *executor) start() {
...
+						case RescheduleMode:
+							return
+						case WaitMode:
+							for {
```
Comment (JohnRoesler) [inline executor.go:56]: realized I need a way to stop this for loop if the scheduler is stopped - working on it

Comment (Streppel) [reply executor.go:56]: oh, just realized we might need to take into consideration job removals as well right?

**Pull request merged**